internal/utils/flags/flags.go

```diff
@@ -243,3 +245,29 @@ func (f StringSliceFlag) Register(fs *pflag.FlagSet) {
 func MarkHidden(fs *pflag.FlagSet, name string) {
 	fs.MarkHidden(name) //nolint: errcheck
 }
+
+// Deprecator is used to mark a flag for deprecation with Message, if To is provided a forwarding message will
+// be generated and used as Message.
+type Deprecator struct {
+	FirstUnsupportedVersion string
+	To                      string
+}
+
+// Deprecate returns true if the flag needs deprecation, and prints deprecation message based on Deprecator information
+// in Meta. It returns false if no deprecation is needed.
+func (m Meta) Deprecate(fs *pflag.FlagSet) bool {
+	if m.Deprecated == nil {
+		return false
+	}
+
+	deprecationMessage := fmt.Sprintf(
+		"it will be removed at version %s",
+		m.Deprecated.FirstUnsupportedVersion)
+
+	if m.Deprecated.To != "" {
+		deprecationMessage = fmt.Sprintf("%s, instead please use '%s'", deprecationMessage, m.Deprecated.To)
+	}
+
+	fs.MarkDeprecated(m.Name, deprecationMessage) //nolint: errcheck
```

Comment:
I still want to see this handled in the same way we handle `func MarkHidden(fs *pflag.FlagSet, name string)`, so that we can isolate the `nolint: errcheck` to that one-liner function as well as include the `TODO` for cleaning that up once we can

Reply:
cool ill basically copy the comment here and replace `hidden` with `deprecated`.